add last used ball

ghoulslash opened the pull request:
Add (configurable) last used ball feature

## Description
- Press "R" on the choose-action menu in battle to throw the last-thrown ball.
- Can be turned off by setting `B_LAST_USED_BALL` to `FALSE`
- Can change the button (or button combo) by changing `B_LAST_USED_BALL_BUTTON`. The graphic will need to be updated though

![last_ball](https://user-images.githubusercontent.com/41651341/123477020-b5766280-d5ba-11eb-847b-42bb0c09dc2c.gif)

LOuroboros commented:
There's merge conflicts that need to be solved :eyes:

LOuroboros commented:
There's more merge conflicts in `src/item_use.c` that need to be solved 👀

ghoulslash commented:
ready for re review

AsparagusEduardo commented on include/global.h:
```diff
@@ -485,7 +485,8 @@ struct SaveBlock2
              u16 regionMapZoom:1; // whether the map is zoomed in
     /*0x18*/ struct Pokedex pokedex;
-    /*0x90*/ u8 filler_90[0x8];
+    /*0x90*/ u16 lastUsedBall;
```
I tested behavior in vanilla SwSh and it doesn't keep the last used ball when saving, instead reverting to the first ball in the pocket. I feel that modifying saveblock is out of scope for BE, so only setting up EWRAM data should be enough.

AsparagusEduardo commented on include/constants/battle_config.h:
```diff
@@ -153,6 +153,10 @@
 
+// Last Used Ball
+#define B_LAST_USED_BALL            TRUE        // If TRUE, the "last used ball" feature from Gen 7 will be implemented
+#define B_LAST_USED_BALL_BUTTON     R_BUTTON    // If last used ball is implemented, this button (or button combo) will trigger throwing the last used ball.
```
It'd be good to add in the config comment that the button graphic isn't updated when changing button combinations.

ghoulslash commented:
ready for re-review.

- I've added separate graphics for the last used ball window based on the button config option (L vs. R)
- As per SwSh behavior, I've removed the saveblock field and replaced it with an EWRAM variable.

AsparagusEduardo commented on src/battle_interface.c:
```diff
@@ -3128,3 +3144,202 @@ static void Task_FreeAbilityPopUpGfx(u8 taskId)
+{
+    #if B_LAST_USED_BALL == TRUE
+    if (gLastThrownBall == 0)
+        gLastThrownBall = ITEM_POKE_BALL;
```
Regarding SwSh behavior, when loading a save it loads the first ball in the pocket instead of defaulting to Poké Balls. Could you implement this?

AsparagusEduardo reviewed (changes requested):
I have a suggestion: : a ball count for the interface could be useful for this feature :D
If you don't want to do it it's ok :)

AsparagusEduardo commented:
Calls to `lastUsedBall` were removed, but it's still in the saveblock.
